akka-stream-tests/src/test/scala/akka/stream/scaladsl/SourceSpec.scala

```diff
@@ -272,6 +272,20 @@ class SourceSpec extends AkkaSpec with DefaultTimeout {
     }
   }
 
+  "Cycle Source" must {
+
+    "continuously generate the same sequence" in {
+      val expected = Seq(1, 2, 3, 1, 2, 3, 1, 2, 3)
+      Source.cycle(1, 2, 3).grouped(9).runWith(Sink.head).futureValue should ===(expected)
+    }
+
+    "reject creation from empty sequence" in {
+      intercept[IllegalArgumentException] {
+        Source.cycle()
```

Comment:
No, as I understand it, the suggestion from @viktorklang is to not use varargs, but:

    def cycle[O](f: function.Creator[java.util.Iterator[O]]): javadsl.Source[O, NotUsed]

similar for scaladsl.

However, I might have misunderstood the purpose and the use case motivates varargs?